packages/@react-stately/list/src/useListState.ts

```diff
@@ -34,6 +34,16 @@ export interface ListState<T> {
   selectionManager: SelectionManager
 }
 
+function getNextLogicalFocusKeys<T extends object>(collection: ListCollection<T>, focusedKey: Key): Array<Key> {
+  const nextFocusKeys: Array<Key> = [];
+  let nextFocusKey = collection.getKeyAfter(focusedKey);
+  while (nextFocusKey) {
+    nextFocusKeys.push(nextFocusKey);
+    nextFocusKey = collection.getKeyAfter(nextFocusKey);
+  }
```

Comment:
matches init; test; incr structure of a for-loop:

```suggestion
  for (let nextFocusKey = collection.getKeyAfter(focusedKey);
      nextFocusKey;
      nextFocusKey = collection.getKeyAfter(nextFocusKey))
  {
    nextFocusKeys.push(nextFocusKey);
  }
```